base/doc/ltnews33.tex

```diff
@@ -503,6 +503,33 @@ \subsection{Small fix for rolling back prior to 2020-02-02}
 
 
 
+\subsection{Add \cs{tracingstacklevels}
+            and \cs{tracinglostchars}\texttt{=3} to \cs{tracingall}}
+
+In July 2020 David Jones suggested an extension to \TeX{} engines, that
+added the possibility to set \cs{tracinglostchars}\texttt{=3} to have an
+error in case some character is missing from a font.  In previous years,
+the warning for a missing character would be silently printed to the
+\texttt{.log} file (if $\cs{tracinglostchars}>0$) and to the terminal
+(if ${}>1$).  This extension was added for \TL{} and \MiKTeX{} (except
+in Knuth's \TeX, of course) and now with $\cs{tracinglostchars}>2$ you
+get an error on a missing glyph.
+
+Later, in January 2021, Petr Olšák suggested yet another extension, a
+new primitive parameter \cs{tracingstacklevels} that, when positive (and
+when \cs{tracingmacros} is also positive), will print a visual
+indication of the macro nesting level in \TeX's tracing information.
+
+Both these changes were incorporated to \LaTeX's debugging macros
+\cs{tracingall} and \cs{tracingnone}, so when you use them, the new
+extensions are automatically activated/deactivated if your \TeX{}
+distributions has a recent enough engine.  An example document
+demonstrating these parameters is available in the linked GitHub issue.
```

Comment:
I'm not sure what the best strategy is for \tracinglostchars. It seems to me that for that parameter there is not a "single" default, i.e., you may want to unconditionally set it to 3 for a document.
In that case \traceoff or \tracingnone should not kill that setting, should it?

Reply:
Hm... Indeed!  Maybe save the current value of `\tracinglostchars` in `\tracingall` and restore in `\tracingnone`?

Reply:
It is a general problem also with other tracing parameters, e.g., if you set \showoutput and then somewhere you go \traceon ... \traceoff or \tracingnone settings arge gone. So perhaps it does need a general solution saving and restoring all values. We could implement something like that now or as a separate issue at some point in the future  (maybe that is better).

Reply:
An approach similar to what `\ExplSyntaxOn/Off` do with catcodes should do:
```latex
\let\tracingnone\relax
\def\tracingall{%
  \edef\tracingnone{%
    \tracinglostchars=\the\tracinglostchars\relax
    \tracingmacros=\the\tracingmacros\relax
    ...
    \let\tracingnone\relax
  }
  \tracinglostchars=\thr@@
  \tracingmacros=\tw@
  ...
}
```

Then `\tracingnone` would normally be a no-op, and `\tracingall` would define `\tracingnone` to do the proper resetting.

I'll leave it as that for now and open a separate issue for this, to keep things well separated